src/renderer/runner.ts

```diff
@@ -203,8 +204,17 @@ export class Runner {
    */
   public stop(): boolean {
     this.appState.isRunning = !!this.child && !this.child.kill();
-    const isIdle = !this.appState.isRunning;
-    return isIdle;
+
+    // If the child process is still alive 1 second after we've
+    // attempted to kill it by normal means, kill it forcefully.
+    setTimeout(() => {
+      const pid = this.child?.pid;
+      if (pid && isRunning(pid)) {
+        this.child?.kill('SIGKILL');
+      }
+    }, 1000);
```

Comment:
This makes the return value of this function kind of weird, since the true state could change right after this value is returned. Maybe this should return a promise so that users can await the stop iff they want, and just invoke without awaiting if that's all they need.

Also, it looks like the boolean being returned is not checked anywhere, so Maybe this should return `Promise<void>`

Reply:
this is just used in the click handler for stop/start, so i think we can just......remove the return value maybe 🤔

Reply:
I've a slight preference for `Promise<void>` because that's what I would expect to see from a method named "stop()" but... **nobody currently calling this function cares** so YAGNI wins! :smile_cat: